packages/slate-editor/src/modules/editor-v4-rich-formatting-menu/RichFormattingMenu.tsx

```diff
@@ -61,14 +65,45 @@ export const RichFormattingMenu: FunctionComponent<Props> = ({
     }
 
     const show = isSelectionSupported(editor);
-    const activeNodeType = getRichFormattingBlockNodeType(editor);
     const [href, setHref] = useState<string>('');
     const [isExistingLink, setIsExistingLink] = useState<boolean>(false);
     const [linkCandidateId, setLinkCandidateId] = useState<string | null>(null);
     const [linkPath, setLinkPath] = useState<Path | null>(null);
     const [savedSelection, setSavedSelection] = useState<Path | Range | null>(null);
     const linkCandidateElement = useLinkCandidateElement(linkCandidateId);
 
+    const alignment = EditorCommands.getAlignment(editor, defaultAlignment);
+    const isBoldActive = EditorCommands.isMarkActive(editor, MarkType.BOLD);
+    const isItalicActive = EditorCommands.isMarkActive(editor, MarkType.ITALIC);
+    const isUnderlineActive = EditorCommands.isMarkActive(editor, MarkType.UNDERLINED);
+    const isSuperScriptActive = EditorCommands.isMarkActive(editor, MarkType.SUPERSCRIPT);
+    const isSubScriptActive = EditorCommands.isMarkActive(editor, MarkType.SUBSCRIPT);
+    const isLinkActive = EditorCommands.isBlockActive(editor, LINK_NODE_TYPE);
+    const formatting = getCurrentFormatting(editor);
+
+    function handleSubSupClick() {
+        if (isSuperScriptActive) {
+            EditorCommands.toggleMark(editor, MarkType.SUPERSCRIPT);
+            EditorCommands.toggleMark(editor, MarkType.SUBSCRIPT);
```

Comment:
Is it correct that we have two toggleMark in a row?

Reply:
Yes. It was this way before the refactoring.